index.js

```diff
@@ -1,102 +0,0 @@
-const readline = require("readline");
-const rl = readline.createInterface(process.stdin, process.stdout);
-
-start();
-
-async function start() {
-  console.log("Choose your game!!");
-  let game = "false";
-
-  function ask(questionText) {
-    return new Promise((resolve, reject) => {
-      rl.question(questionText, resolve);
-    });
-  }
-
-  game = await ask("reverse or guess? ");
-  if (game === "guess") {
-    let minNumber = 1;
-//Original Start
-    let minCurrent = 0;
-    ("yes");
-    while (playAgain === "yes") {
-      console.log(
-        "Welcome to the game, please pick a max number for me to guess!"
-      );
-      let maxNumber = await ask(" What do you choose for your max number?");
-      let medianChange = maxNumber / 2;
-      let maxCurrent = maxNumber;
-
-      // High and Low Function
-      let Guess = await ask("Is " + medianChange + " your number?");
-
-      if (Guess === "yes") {
-        console.log("Congrats, I won!");
-        playAgain = await ask("Would you like to play again?");
-      } else {
-        while (Guess !== "yes") {
-          // let me know I'm in loop 
-          //console.log("im in" )
-          let currentGuess = await ask("higher or lower?");
-          console.log("\ncurrent guess is " + medianChange);
-         
-          if (currentGuess === "h" || currentGuess === "higher") {
-            //Debugger viewing current variable values
-            // console.log("This is maxnumber" + maxNumber);
-            // console.log("This is current" + maxCurrent);
-            // console.log("This is median" + medianChange);
-            // console.log("im high");
-            medianChange = Math.round((+maxCurrent + medianChange) / 2);
-          } else if (currentGuess === "l" || currentGuess === "lower") {
-            //Debugger 2
-            // console.log("This is min number" + minNumber);
-            // console.log("This is current" + minCurrent);
-            // console.log("This is median" + medianChange);
-            // console.log("im low");
-            medianChange = Math.round((+minCurrent + medianChange) / 2);
-          }
-
-          await ask("Is " + medianChange + " your number?");
```

Comment:
Once I was able to get into this loop I could also enter into each of these two checks. I noticed you have some debugging going on here to be able to see your variables as they are modified already. This is a great thing to do. Once you are able to get the numbers sorted you can set the new high/low range accordingly.  I was able to get a random number returned each time I answered in this current state, it just did not always return a random number in the correct range / direction.